### PR TITLE
split aws code from platform

### PR DESCRIPTION
**Pull request by mischief**

### CODE REVIEW

File: platform/machine/aws/machine.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+func (am *machine) PasswordSSHClient(user string, password string) (*ssh.Client, error) {
+	return am.cluster.PasswordSSHClient(am.IP(), user, password)
+}
+
+func (am *machine) SSH(cmd string) ([]byte, error) {
+	return am.cluster.SSH(am, cmd)
+}
+
+func (am *machine) Destroy() error {
+	// XXX: defer DelMach here?
```
Comment (marineam) [inline platform/machine/aws/machine.go]: Why?

File: platform/api/aws/ec2.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// CreateInstances creates EC2 instances with a given image ID, ssh key name, user data, instance type, and security group. If wait is true, CreateInstances will block until all instances are reachable by SSH.
+func (a *API) CreateInstances(imageid, keyname, userdata, instancetype, securitygroup string, count uint64, wait bool) ([]*ec2.Instance, error) {
```
Comment (marineam) [inline platform/api/aws/ec2.go]: This many parameters is pretty awkward, maybe just accept RunInstancesInput? Or a similar but simplified struct.

**Comment by marineam** [comment]
Other than CreateInstances, looking good.

File: platform/api/aws/ec2.go (hunk 2 above)
Comment (mischief) [inline platform/api/aws/ec2.go]: id like to hide the aws api as much as possible. i'm considering using pflag.FlagSet for all of the optiony things..

**Comment by mischief** [comment]
ptal

**Comment by marineam** [comment]
LGTM

**Pull request merged**